### ISSUE TITLE
Task header removed prematurely

### ISSUE THREAD

**Issue by mfranciszkiewicz** [golemfactory/clay#2344]
The error occurs in `TaskComputer.__compute_task`:

`Subtask '132132097823085547139854912857353388696' of task '6782dcf2-a6aa-4ded-8438-cccffb9bebc7' cannot be computed: task header has been unexpectedly removed`

TaskHeader, for which a subtask definition has been given, is removed before computation starts.

**Comment by gcarq** [comment]
Reproducible with 97af5813bf7f3baf561dc2344ea1a8df4a54329d using mainnet.

Below is the output grepped for the failed task: `b51532ec-47f8-11e8-a492-059d9ca1500a`

```
Apr 24 21:50:27 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 21:52:21 serv13580311 golemapp[4676]: INFO     [golem.task.taskserver              ] Cannot connect to task b51532ec-47f8-11e8-a492-059d9ca1500a owner
Apr 24 21:52:21 serv13580311 golemapp[4676]: INFO     [golem.task.taskserver              ] Removing task b51532ec-47f8-11e8-a492-059d9ca1500a from task list
Apr 24 21:52:21 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: Connection failed
Apr 24 21:53:32 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NoMoreSubtasks
Apr 24 21:53:32 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NoMoreSubtasks
Apr 24 21:55:25 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 21:56:00 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NoMoreSubtasks
Apr 24 21:56:29 serv13580311 golemapp[4676]: WARNING  [golem.task.taskcomputer            ] Subtask 'f82b971a-47f8-11e8-aac9-059d9ca1500a' of task 'b51532ec-47f8-11e8-a492-059d9ca1500a' cannot be computed: task header has been unexpectedly removed
Apr 24 21:59:08 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 21:59:14 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NoMoreSubtasks
Apr 24 22:05:17 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:05:20 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:09:00 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:17:09 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:20:10 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:20:45 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:23:46 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:24:01 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:25:49 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Removing comp_task after deadline: b51532ec-47f8-11e8-a492-059d9ca1500a
Apr 24 22:27:07 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:27:17 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:30:19 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:31:01 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:34:04 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:34:09 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:37:13 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:37:19 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:40:26 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:40:51 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:43:56 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:44:02 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:47:08 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:47:34 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:50:40 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:51:06 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:54:12 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:54:14 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 22:57:15 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 22:57:44 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:00:46 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:00:50 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:02:25 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Removing comp_task after deadline: b51532ec-47f8-11e8-a492-059d9ca1500a
Apr 24 23:03:51 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:04:08 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:07:10 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:07:14 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:10:22 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:10:24 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:13:32 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:13:54 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:16:55 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:17:31 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:20:38 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:20:43 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:23:51 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:23:53 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:27:00 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:27:08 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:30:10 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:30:31 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:33:33 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:33:40 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:36:42 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:37:01 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:39:16 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Removing comp_task after deadline: b51532ec-47f8-11e8-a492-059d9ca1500a
Apr 24 23:40:04 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:40:07 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:43:08 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:43:12 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:46:15 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:46:23 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:49:28 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:49:33 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:52:39 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:53:43 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 24 23:56:52 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 24 23:57:02 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 25 00:00:12 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 25 00:15:15 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Removing comp_task after deadline: b51532ec-47f8-11e8-a492-059d9ca1500a
Apr 25 00:20:23 serv13580311 golemapp[4676]: INFO     [golem.task.taskserver              ] Cannot connect to task b51532ec-47f8-11e8-a492-059d9ca1500a owner
Apr 25 00:20:23 serv13580311 golemapp[4676]: INFO     [golem.task.taskserver              ] Removing task b51532ec-47f8-11e8-a492-059d9ca1500a from task list
Apr 25 00:20:23 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: Connection failed
Apr 25 00:23:31 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 25 00:23:34 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 25 00:26:43 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 25 00:26:45 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 25 00:29:46 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 25 00:29:50 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 25 00:32:53 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 25 00:33:19 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 25 00:36:27 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 25 00:36:40 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 25 00:39:42 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Adding task 'b51532ec-47f8-11e8-a492-059d9ca1500a' support=<SupportStatus ok ({})>
Apr 25 00:50:36 serv13580311 golemapp[4676]: INFO     [golem.task.taskcomputer            ] Task b51532ec-47f8-11e8-a492-059d9ca1500a request rejected: REASON.NotMyTask
Apr 25 00:50:36 serv13580311 golemapp[4676]: INFO     [golem.task.taskkeeper              ] Removing comp_task after deadline: b51532ec-47f8-11e8-a492-059d9ca1500a

```

**Comment by maaktweluit** [comment]
Resolved by #4159
